dso_pkg/dso/language_model/language_model_prior.py

```
@@ -49,15 +49,23 @@ def __init__(self, dso_library,
         self._zero_state = np.zeros(num_hidden, dtype=np.float32)
 
     def load_model(self, saved_language_model_path):
-        sess = tf.compat.v1.Session()
-        saver = tf.train.Saver()
-        saver.restore(sess,tf.train.latest_checkpoint(saved_language_model_path))
-        return sess
+        """Load pre-trained model using TF2."""
+        # Try to load as SavedModel first
+        try:
+            self.language_model = tf.keras.models.load_model(saved_language_model_path)
+        except:
```

Ruff (0.12.2)

Do not use bare `except`

(E722)